### DISCUSSION TITLE
Role & Permission ( No Model )

### DISCUSSION THREAD

**Discussion by emresasi** [thedevdojo/voyager#5459] [Q&A]
Hello, I have a problem but couldn't find a solution. I have a page with only 1 controller. This page does not have a db table and model. In this controller, the queries I want are running and giving me some reports. I can't naturally create a bread for these pages. So I can't see it on the role page. When I add permission manually, I see it on the role page, but it only shows and hides it in the menu, it has no effect on the authority of the page and I can enter the page.
When I add $this->authorize('browse_xxx') to the relevant function, it does not accept a permission named "browse_xxx", it says you are not authorized to login. How can I authorize a page that has no model and no table?
![1](https://user-images.githubusercontent.com/6942826/136369840-fb9a5ecd-a98a-4372-83bb-781d96c53060.png)
![3](https://user-images.githubusercontent.com/6942826/136370182-08f96cd5-08be-4835-994e-04cd9682119a.png)
![4](https://user-images.githubusercontent.com/6942826/136370194-a936c7c4-a864-4b50-8de7-d8150cf40fed.png)

**Comment by fletch3555** [comment]
`$this->authorize(...)` requires that a gate/policy exists with that name.  Voyager automatically registers policy classes for each of the configured BREAD types.  As for these non-BREAD permissions, Voyager automatically [registers a gate for the built-in ones](https://github.com/the-control-group/voyager/blob/1.5/src/VoyagerServiceProvider.php#L305-L309), but doesn't know anything about ones you create manually.

In short, you're just missing the wiring between the permission in the database and the authorization check in your controller.
